process/smartContract/process.go

```diff
@@ -1290,6 +1293,13 @@ func (sc *scProcessor) ProcessIfError(
 
 	sc.txFeeHandler.ProcessTransactionFee(consumedFee, big.NewInt(0), txHash)
 
+	if sc.flagOptimizeNFTStore.IsSet() {
+		err = sc.blockChainHook.SaveNFTMetaDataToSystemAccount(tx)
```

Comment:
is it ok to be called on ProcessIfError?

Reply:
yes. It must be called here - as we have to save the NFT metadata to the system account if the cross shard NFT transfer failed for whatever reason. As we will not send this information the second time.

Reply:
ok!